apps/company/serializers.py

```
@@ -248,34 +308,595 @@ class Meta:
+    shelf_name = serializers.CharField()
+    stock = serializers.IntegerField()
+
+    class Meta:
+        model = Shelf
+        fields = ['product','shelf_name','stock']
+
+    def update(self, instance: Shelf, validated_data):
+        
+        try:
+            product = Product.objects.get(vendor_code=validated_data['product'])
+        except:
```

Ruff

Do not use bare `except`

(E722)